### PR TITLE
Retry update on conflict

### PR DESCRIPTION
**Pull request by gregorylegarec**
Allow to retrie update on 409 error :
fetch the actual document, merge it with the data we want to update, and try to perform the update again.

### CODE REVIEW

File: packages/cozy-stack-client/src/DocumentCollection.js [hunk 1]
```diff
@@ -180,12 +180,30 @@ class DocumentCollection {
     }
   }
 
-  async update(document) {
-    const resp = await this.stackClient.fetchJSON(
-      'PUT',
-      uri`/data/${this.doctype}/${document._id}`,
-      document
-    )
+  async update(document, options = {}) {
```
Comment (Crash--) [inline packages/cozy-stack-client/src/DocumentCollection.js]: can you add a js doc about `options`

File: packages/cozy-stack-client/src/DocumentCollection.js [hunk 2]
```diff
@@ -180,12 +180,30 @@ class DocumentCollection {
...
+    let resp
+    try {
+      resp = await this.stackClient.fetchJSON(
+        'PUT',
+        uri`/data/${this.doctype}/${document._id}`,
+        document
+      )
+    } catch (error) {
+      const { status } = error
+      if (status === 409 && retriesOnConflict) {
```
Comment (edas) [inline packages/cozy-stack-client/src/DocumentCollection.js]:
```suggestion
      if (status === 409 && retriesOnConflict && retriesOnConflict > 0) {
```

**Comment by ptbrowne** [comment]
Apart from the suggestion from edas, the implementation seems OK. Please add test though.

File: packages/cozy-stack-client/src/DocumentCollection.js (hunk 2 above)
Comment (CPatchane) [reply packages/cozy-stack-client/src/DocumentCollection.js]: A number less or equal than 0 isn't considered as a falsy value?

Comment (edas) [reply packages/cozy-stack-client/src/DocumentCollection.js]: `if (-1) { console.log("true") } else { console.log("false") }`
--> "true"

Comment (CPatchane) [reply packages/cozy-stack-client/src/DocumentCollection.js]: OK I was wrong, thanks 👍

Comment (gregorylegarec) [reply packages/cozy-stack-client/src/DocumentCollection.js]: Only `0` is falsy.

**Comment by gregorylegarec** [comment]
I think this method has eventually too much impact, and shoud propbably not be done at this level. Let's think on it a little bit more before proposing a better solution.

**Pull request closed without merging**